File: abc/081/b/java/Main.java
import java.util.List;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            List<Integer> numList = scanner.tokens()
                    .skip(1)
                    .map(Integer::parseInt)
                    .toList();

            int numListOr = numList.stream()
                    .reduce((a, b) -> a | b).get();

            int count = 0;
            while (numListOr % 2 == 0) {
                numListOr /= 2;
                count++;
            }

            System.out.println(count);
        }
    }
}

File: abc/081/b/python/main.py
from functools import reduce

length = int(input())
num_list = list(map(int, input().split()))

num_list_or = reduce(lambda x, y: x | y, num_list)

count = 0
while num_list_or % 2 == 0:
    count += 1
    num_list_or = num_list_or // 2

print(count)
